SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -85,13 +90,35 @@ - (void)sdl_stopEventListening {
     [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
+- (void)sdl_backgroundTaskStart {
+    if (self.backgroundTaskId != UIBackgroundTaskInvalid) {
+        return;
+    }
+    
+    self.backgroundTaskId = [[UIApplication sharedApplication] beginBackgroundTaskWithName:@"SDLIAPConnectionLoop" expirationHandler:^{
+        [self sdl_backgroundTaskEnd];
+    }];
+}
+
+- (void)sdl_backgroundTaskEnd {
+    if (self.backgroundTaskId != UIBackgroundTaskInvalid) {
+        [[UIApplication sharedApplication] endBackgroundTask:self.backgroundTaskId];
+        self.backgroundTaskId = UIBackgroundTaskInvalid;
+    }
+}
+
+
 #pragma mark - EAAccessory Notifications
 
 - (void)sdl_accessoryConnected:(NSNotification *)notification {
     NSMutableString *logMessage = [NSMutableString stringWithFormat:@"Accessory Connected, Opening in %0.03fs", self.retryDelay];
     [SDLDebugTool logInfo:logMessage withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
 
     self.retryCounter = 0;
+    if ([UIApplication sharedApplication].applicationState == UIApplicationStateBackground) {
+        [self sdl_backgroundTaskStart];
```

Comment:
Instead of monitoring background and starting the task then, we should just wrap our connection in a background task, in other words, always start a background task before we are trying to connect, and always cancel it after we've connected / dealloc. There's no cost to starting background tasks, and it'll just automatically handle it when we go into the background. Then all of the background monitoring code can be removed.